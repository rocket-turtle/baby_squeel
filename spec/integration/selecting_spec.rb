require 'spec_helper'

describe '#selecting' do
  it 'selects using arel' do
    relation = Post.selecting { [id, title] }
    expect(relation).to match_sql_snapshot
  end

  it 'selects associations' do
    relation = Post.joins(:author).selecting {
      [id, author.id]
    }

    expect(relation).to match_sql_snapshot
  end

  it 'selects aggregates' do
    relation = Post.selecting { id.count }

    expect(relation).to match_sql_snapshot
  end

  it 'selects using operations' do
    relation = Post.joins(:author).selecting {
      author.id - id
    }

    expect(relation).to match_sql_snapshot
  end

  it 'selects on an aliased table' do
    relation = Post.joins(author: :posts).selecting {
      author.posts.id
    }

    expect(relation).to match_sql_snapshot
  end
end
