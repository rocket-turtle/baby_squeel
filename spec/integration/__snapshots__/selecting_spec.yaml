---
"#selecting selects using arel 1": SELECT "posts"."id", "posts"."title" FROM "posts"
"#selecting selects associations 1": SELECT "posts"."id", "authors"."id" FROM "posts"
  INNER JOIN "authors" ON "authors"."id" = "posts"."author_id"
"#selecting selects aggregates 1": SELECT COUNT("posts"."id") FROM "posts"
"#selecting selects using functions 1": SELECT coalesce("posts"."id", "authors"."id")
  FROM "posts" INNER JOIN "authors" ON "authors"."id" = "posts"."author_id"
"#selecting selects using operations 1": SELECT ("authors"."id" - "posts"."id") FROM
  "posts" INNER JOIN "authors" ON "authors"."id" = "posts"."author_id"
"#selecting selects on an aliased table 1": SELECT "posts_authors"."id" FROM "posts"
  INNER JOIN "authors" ON "authors"."id" = "posts"."author_id" INNER JOIN "posts"
  "posts_authors" ON "posts_authors"."author_id" = "authors"."id"
"#selecting selects on an aliased table 1 (Active Record: v8.1)": SELECT "posts_authors"."id"
  FROM "posts" INNER JOIN "authors" ON "authors"."id" = "posts"."author_id" INNER
  JOIN "posts" AS "posts_authors" ON "posts_authors"."author_id" = "authors"."id"
